modules/layers/src/arc-layer/arc-layer.ts

```diff
@@ -257,28 +259,12 @@ export default class ArcLayer<DataT = any, ExtraPropsT extends {} = {}> extends
 
   protected _getModel(): Model {
     const {numSegments} = this.props;
-    let positions: number[] = [];
-    /*
-     *  (0, -1)-------------_(1, -1)
-     *       |          _,-"  |
-     *       o      _,-"      o
-     *       |  _,-"          |
-     *   (0, 1)"-------------(1, 1)
-     */
-    for (let i = 0; i < numSegments; i++) {
-      positions = positions.concat([i, 1, 0, i, -1, 0]);
-    }
 
     const model = new Model(this.context.device, {
       ...this.getShaders(),
       id: this.props.id,
       bufferLayout: this.getAttributeManager()!.getBufferLayouts(),
-      geometry: new Geometry({
-        topology: 'triangle-strip',
-        attributes: {
-          positions: {size: 3, value: new Float32Array(positions)}
-        }
-      }),
+      topology: 'triangle-strip',
       isInstanced: true
     });
```

Comment:
Maybe two parts here:

1. model without geometry
2. model without vertex count

For me (2) was the more surprising part, since that would seem to be an invalid initialization state.  If a Model without Geometry or Attributes is valid, is a Geometry without Attributes also valid? Not suggesting we add more flexibility, just maybe clarity to make valid/invalid states obvious.

Reply:
Yes... I am not sure how to encode these things in APIs and types. 

"Either-or type checks" require a fair bit of typescript machinery (discriminated unions) that makes code less readable.

Reply:
Added `vertexCount` to model construction

Reply:
The `geometry` & `vertexCount` props are quite closely linked. `Model._setGeometryAttributes()` sets the `vertexCount` based on the number of vertices obtained from the `gpuGeometry`, which is then optionally overwritten with `props.vertexCount`. I wonder if it would be better to support a `NullGeometry` that contained nothing but a `vertexCount` and then we remove the `vertexCount` prop altogether and make `props.geometry` required.

Reply:
I like the general idea... Though we do have a similar situation with `instanceCount` so it would be nice to handle both in a similar way.

Something I really liked was the design we did in deck.gl-native, where we took the "table view" and avoided the term geometry and instead said that a model can have two tables, basically something like:

```
model.setAttributes(table: arrow.Table) 
model.setInstancedAttributes(table: arrow.Table)
```

This makes it clear that these are tables of different lengths, and the table lengths determine the vertexCount and instanceCount (though these can be overridden during draw of course, to just draw a range).

Of course we can still have a convenience function setGeometries that contain table, indices and topology and that calls this underlying function.

We  dohave the complication of indexed geometries and vertex-sharing topologies to deal with (though modern GPU programming practices often seem to recommend avoiding indices and vertex-sharing topologies, so we could make the decision to start phasing that out).

Reply:
My feeling is that indexed geometries are still very much recommended, and are key to certain optimizations. I agree that alternative geometries like strips and lists could be phased out.

While working on porting code to v9, I personally found the number of different ways to pass geometry into Model confusing, as I often wasn't sure if I was doing it the correct way, or mixing two correct ways incorrectly. That can be a hidden cost of convenience methods.

I quite like the "two tables" idea!

(and sorry for starting a long thread here, @Pessimistress please feel free to merge!)